src/base/Common.cpp

```diff
@@ -1154,7 +1154,8 @@ UnicodeString ApiPath(UnicodeString APath)
 {
   UnicodeString Result = APath;
 
-  if (IsWin7() || (Result.Length() >= MAX_PATH))
+  // Max path for directories is 12 characters shorter than max path for files
+  if (IsWin7() || (Result.Length() >= MAX_PATH - 12))

```

Comment:
Do we need this condition at all?

There is no harm in always using `\\?\` paths and it would be more consistent.
Current implementation can lead to bizarre scenarios like "copying a folder named com1 only works in Windows 7, but not in older or newer OSes".